tilavarauspalvelu/integrations/email/main.py

```diff
@@ -445,56 +479,74 @@ def send_reservation_requires_payment_email(
-            text_content=render_text(email_type=email_type, context=context),
-            html_content=render_html(email_type=email_type, context=context),
-        )
+        email = EmailData.build(recipients, context, email_type)
+        send_emails_in_batches_task.delay(email_data=email)
 
     @staticmethod
-    def send_reservation_rejected_email(
-        reservation: Reservation,
+    def send_seasonal_reservation_modified_series_email(
+        reservation_series: RecurringReservation,
         *,
         language: Lang | None = None,
     ) -> None:
-        """
-        Sends an email about the reservation being rejected in handling.
+        # Do not send the email, if the reservation series not attached to an allocated time slot.
+        if reservation_series.allocated_time_slot is None:
+            return
 
-        :param reservation: The reservation the email concerns.
-        :param language: The language of the email. Determine from reservation if not given.
-        """
-        # Prevent accidental sending of wrong email.
-        if reservation.state != ReservationStateChoice.DENIED:
+        reservation: Reservation | None = reservation_series.reservations.last()
+        if reservation is None:
+            return
+        if reservation.type != ReservationTypeChoice.SEASONAL:
+            return
+        if reservation_series.actions.get_application_section() is None:
             return
 
-        # Only send emails if reservation is not in the past.
-        if local_datetime() >= reservation.end.astimezone(DEFAULT_TIMEZONE):
+        recipients = get_reservation_email_recipients(reservation=reservation)
+        if not recipients:
+            SentryLogger.log_message(
+                "No recipients for reservation series modified email",
+                details={"reservation_series": reservation_series.pk},
+            )
             return
 
-        # Don't send notifications for reservations not made by the reservee themselves.
-        if reservation.type != ReservationTypeChoice.NORMAL:
+        if language is None:
+            language = get_reservation_email_language(reservation=reservation)
+
+        email_type = EmailType.SEASONAL_RESERVATION_MODIFIED_SERIES
+        context = get_context_for_seasonal_reservation_modified_series(reservation_series, language=language)
+        email = EmailData.build(recipients, context, email_type)
+        send_emails_in_batches_task.delay(email_data=email)
+
+    @staticmethod
+    def send_seasonal_reservation_rejected_series_email(
+        reservation_series: RecurringReservation,
+        *,
+        language: Lang | None = None,
+    ) -> None:
+        reservation: Reservation | None = reservation_series.reservations.last()
+        if reservation is None:
+            return
+        if reservation.type != ReservationTypeChoice.SEASONAL:
+            return
+        if reservation_series.actions.get_application_section() is None:
```

Comment:
I would prefer this check that you have in `send_seasonal_reservation_modified_series_email`:

```python
if reservation_series.allocated_time_slot is None:
    return
```